src/interaction/select.js

```diff
@@ -375,7 +375,10 @@ function handleEvent(mapBrowserEvent) {
          * @return {boolean|undefined} Continue to iterate over the features.
          */
         function(feature, layer) {
-          if (this.filter_(feature, layer)) {
+          /* If feature layer is null, omit the selection.
+           * Prevents mistakenly selecting features from the overlay.
+           */
+          if (this.filter_(feature, layer) && layer) {
```

Comment:
Tiny perfomance suggestion:
```suggestion
          if (layer && this.filter_(feature, layer)) {
```